doc: release_notes: skim down the release notes content

fabiobaltieri opened the pull request:
Skim down the release notes file, clarify what goes there and more importantly what doesn't

---
Demo: https://builds.zephyrproject.io/zephyr/pr/82542/docs/releases/release-notes-4.1.html

This is a proposal to skim down the release notes, main idea:

- keep removed/deprecated APIs, although arguably the removed one should be in the release notes
- list all new driver, samples, boards, APIs
  - just link the thing, all of those should already have a documentation generated from the code, duplicating it here is a waste of time and energy for something that lives and dies in the specific release file, that effort should be put in the corresponding documentation in the code instead, initial idea is to keep these lists generated but then maybe refine them further, group, something like that
  - no bugfixes or minor changes, it's called release notes, not changelog
- leave a space for "notable features" and try to have it for stuff that does not fall in the categories above, especially if it does not have a description somewhere else in the code already

TODO: some of the stuff that is still there needs to go in the migration guide, will move it at a later stage

pdgendt commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -24,351 +47,99 @@ https://docs.zephyrproject.org/latest/security/vulnerabilities.html
+New Boards & SoC
+****************
+..
+  Just link the board, detals go in the board description
```
`s/detals/details/` multiple times

fabiobaltieri replied:
hey at least I was consistent

fabiobaltieri committed:
doc: release_notes: skim down the release notes content

Skim down the release notes file, clarify what goes there and more
importantly what doesn't.

Add auto-generated lists of new boards/samples/drivers.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>
Co-authored-by: Benjamin Cabé <benjamin@zephyrproject.org>

jhedberg commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -24,359 +47,177 @@ https://docs.zephyrproject.org/latest/security/vulnerabilities.html
+============
 
- * The deprecated Bluetooth HCI driver API has been removed. It has been replaced by a
+ * The Bluetooth HCI driver API has been removed. It has been replaced by a
```
I feel like this should still say something like "The old HCI driver API", since we still have an HCI driver API (described in the next sentence).

fabiobaltieri replied:
yeah I was trying to remove some wording duplication, reading this back I made this sound super weird, changed it to "The legacy Bluetooth HCI..."

aescolar reviewed (commented):
+1

dkalowsk commented:
Overall a nice pair down of the notes.  I keep thinking that maybe the per-driver and per-subsys should just be a prepopulated GitHub query link such that anyone can go see the change log for the section.

cfriedt reviewed (dismissed):
LGTM

fabiobaltieri commented:
> Overall a nice pair down of the notes. I keep thinking that maybe the per-driver and per-subsys should just be a prepopulated GitHub query link such that anyone can go see the change log for the section.

yeaah good luck with that, it's somewhat easy to generate though at least, if we are happy with the format I'll work on some helper next... somewhat ironic that I wrote to avoid dumping generated list and then half of the page is generated. :-)

henrikbrixandersen commented:
Please rebase on `main` and resolve any merge conflicts.

fabiobaltieri commented:
yeah I was waiting for Ben to confirm he's fine with the format, this conflicts continuously, @kartben do you think the general form is alright here?

kartben commented:
> yeah I was waiting for Ben to confirm he's fine with the format, this conflicts continuously, @kartben do you think the general form is alright here?

have a few pending comments -- need to come back to this early this week, sorry for having delayed/blocked this for so long

kartben commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -1,5 +1,28 @@
+  migration guide instead. (TODO: move the removed APIs section in the
+  migration guide)
+
+  Could the detailed paragraph that you are about to write go in the actual
+  project documentation instead and just be linked here? Do that instead.
```
this deserves being moved higher up IMO -- people sometimes don't realize that they end up writing really good "documentation" that just happens to get a lot less long-term visibility than it should

fabiobaltieri replied:
done

kartben commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -24,359 +47,178 @@ https://docs.zephyrproject.org/latest/security/vulnerabilities.html
+New Boards & SoC
+****************
+..
+  Just link the board, details go in the board description
```
```suggestion
  You may update this list as you contribute a new board during the release cycle, in order for it
  to be visible to people who might be looking at the working draft of the release notes. However,
  note that this list will be recomputed at the time of the release, so you don't *have* to update
  it.
  In any case, just link the board, details go in the board description.
```

fabiobaltieri replied:
done

kartben commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -24,359 +47,178 @@ https://docs.zephyrproject.org/latest/security/vulnerabilities.html
+New Drivers
+***********
+..
+  Just link the driver, details go in the binding description
```
Assuming we'll do the same here: 

```suggestion
    Same as above for boards, this will also be recomputed at the time of the release.
    Just link the driver, details go in the binding description
```

fabiobaltieri replied:
done

kartben commented on doc/releases/release-notes-4.1.rst:
```diff
@@ -24,359 +47,178 @@ https://docs.zephyrproject.org/latest/security/vulnerabilities.html
 
-* Made these board changes:
+..
+  Just link the samples, details go in the sample documentation itself
```
```suggestion
    Same as above for boards and drivers, this will also be recomputed at the time of the release.
    Just link the sample, details go in the sample documentation itself.
```

fabiobaltieri replied:
done

fabiobaltieri commented on doc/releases/migration-guide-4.1.rst:
```diff
@@ -1,5 +1,10 @@
+  What goes here: anything that is likely to require a change in a downstream
+  application using a specific subsystem or driver. Renamed or removed bindings
+  or properties, Kconfig options, etc.
+
```
done

fabiobaltieri commented:
ok reworded the comments, going to rebase next, that's going to be fun

fabiobaltieri commented:
Thanks Ben, looks super

kartben reviewed (approved):
+1 for the 90% part that Fabio did :)
